### ISSUE TITLE
Compatibility with forged carpet?

### ISSUE THREAD

**Issue by nimuck** [Elephant1214/PaperFixes#5]
I'm using [forged carpet](https://www.curseforge.com/minecraft/mc-mods/forgedcarpet), which provides entity tracking, miscellaneous optimizations and bugfixes similar to the original carpet.

I recently found out that it crashes not only with paperfixes
but also with vintagefix's mixin.bugfix.slow_tps_catchup.

There seems to be a conflict in the watchdogfix (one of the many patches provided by forged carpet), but I'm not very smart with code so I'm not sure.
A quick look would be appreciated!

**Comment by Elephant1214** [comment]
Can you send the crash report? If it has the same tick freeze command as modern carpet, I would guess that it's trying to overwrite the same method as VintageFix and PaperFixes, which there's not much that can be done about.

**Comment by Elephant1214** [comment]
The problem is that it's trying to inject into the method PaperFixes and VintageFix overwrite, you'd have to turn the tick loop improvements off entirely to fix it, which I also wouldn't recommend if you don't need the thing from Forged Carpet since it's so much more efficient.

**Comment by nimuck** [comment]
I see. So if I want to use both, I have to somehow disable method injection of forged carpet.
Thank you for quick reply!

**Comment by Elephant1214** [comment]
Yes. There might be config options, I didn't look too hard at anything other than the conflicting mixin, though. I was thinking about backporting the tick commands from Carpet and now modern Minecraft while I was working on the tick loop, I might actually do that now because of this problem.

**Comment by Elephant1214** [comment]
I'll take a look at Beast, but Lithium doesn't play the greatest with mods. As for Mohist, I'm not sure how compatible it is with everything, or if it even changes anything Paper doesn't, but I'll also take a look.
